addons/mail/static/src/models/message/message.js

```diff
@@ -300,6 +300,17 @@ function factory(dependencies) {
             await this.async(() => this.constructor.moderate([this], decision, kwargs));
         }
 
+        /**
+         * Notifies threadViews about this message being just received. This can
+         * allow them adjust their scroll position if applicable.
+         */
+        notifyThreadViewsMessageReceived() {
+            for (const thread of this.threads) {
+                for (const threadView of thread.threadViews) {
+                    threadView.addComponentHint('message-received', { message: this });
+                }
+            }
+        }
```

Comment:
This feels inappropriate as a public method of `mail.message`.
This should stay private concern of `mail.messaging_notification_handler`.